### ISSUE TITLE
SPERG-related CTD in v2.9.6.0

### ISSUE THREAD

**Issue by SteveTownsend** [SteveTownsend/SmartHarvestSE#207]
Nexus message report. https://forums.nexusmods.com/index.php?app=members&module=messaging&section=view&do=showConversation&topicID=4037388&st=20#msg15968823
Possible heap corruption?
```
Unhandled native exception occurred at 0x7FFED1803640 (SmartHarvestSE.dll+63640) on thread 15696!
FrameworkName: NetScriptFramework
FrameworkVersion: 9
FrameworkArchitecture: x64
GameLibrary: SkyrimSE
GameLibraryVersion: 13
ApplicationName: SkyrimSE.exe
ApplicationVersion: 1.5.97.0
VersionInfo: Successfully loaded
Time: 04 Sep 2020 14:47:44.723
Possible relevant objects (6)
{
  [   1]    TESNPC(Name: `Marcus`, FormId: 00000007, File: `NPC Overrides 2020-08-31.esp <- Realm of Lorkhan - Custom Alternate Start - Choose your own adventure.esp <- AI Overhaul.esp <- Skyrim.esm`)
  [   1]    PlayerCharacter(FormId: 00000014, BaseForm: TESNPC(Name: `Marcus`, FormId: 00000007, File: `NPC Overrides 2020-08-31.esp <- Realm of Lorkhan - Custom Alternate Start - Choose your own adventure.esp <- AI Overhaul.esp <- Skyrim.esm`))
  [   1]    BGSKeyword(Name: VendorItemGem, FormId: 000914ED, File: `Skyrim.esm`)
  [   2]    BGSKeyword(Name: VendorItemOreIngot, FormId: 000914EC, File: `Skyrim.esm`)
  [   6]    TESObjectLIGH(Name: `Torch`, FormId: 00036343, File: `Skyrim.esm`)
  [ 178]    BSFadeNode(Name: `INV_CloudDistant03_50`)
}
Probable callstack
{
  [0]   0x7FFED1803640     (SmartHarvestSE.dll+63640)    
  [1]   0x7FFED17F5EF8     (SmartHarvestSE.dll+55EF8)    
  [2]   0x7FFED17AD02C     (SmartHarvestSE.dll+D02C)     
  [3]   0x7FFED18B4F83     (SmartHarvestSE.dll+114F83)   
  [4]   0x7FFED1818376     (SmartHarvestSE.dll+78376)    
  [5]   0x7FF689566E9D     (SkyrimSE.exe+1236E9D)         unk_1236E50+4D
  [6]   0x7FFED1820970     (SmartHarvestSE.dll+80970)    
  [7]   0x7FF6895809E9     (SkyrimSE.exe+12509E9)         BSScript::NF_util::NativeFunctionBase::Call_12507F0+1F9
  [8]   0x7FF68956710E     (SkyrimSE.exe+123710E)         unk_12370E0+2E
  [9]   0x7FF688F3CCD3     (SkyrimSE.exe+C0CCD3)          BSSmallBlockAllocator::DeallocateAlignImpl_C0CBE0+F3
  [10]  0x7FF689566E9D     (SkyrimSE.exe+1236E9D)         unk_1236E50+4D
  [11]  0x7FF688F325F2     (SkyrimSE.exe+C025F2)          Memory::Free_C02560+92
  [12]  0x7FF689577A0E     (SkyrimSE.exe+1247A0E)         unk_1247960+AE
  [13]  0x7FF689579BF6     (SkyrimSE.exe+1249BF6)         BSScript::SimpleAllocMemoryPagePolicy::Func4_1249B30+C6
  [14]  0x7FF68956710E     (SkyrimSE.exe+123710E)         unk_12370E0+2E
  [15]  0x7FF68956710E     (SkyrimSE.exe+123710E)         unk_12370E0+2E
  [16]  0x7FF68956710E     (SkyrimSE.exe+123710E)         unk_12370E0+2E
  [17]  0x7FF688F3CCD3     (SkyrimSE.exe+C0CCD3)          BSSmallBlockAllocator::DeallocateAlignImpl_C0CBE0+F3
  [18]  0x7FF689566DDA     (SkyrimSE.exe+1236DDA)         unk_1236D10+CA
  [19]  0x7FF688F3CCD3     (SkyrimSE.exe+C0CCD3)          BSSmallBlockAllocator::DeallocateAlignImpl_C0CBE0+F3
  [20]  0x7FF689566E9D     (SkyrimSE.exe+1236E9D)         unk_1236E50+4D
  [21]  0x7FF688F325F2     (SkyrimSE.exe+C025F2)          Memory::Free_C02560+92
  [22]  0x7FF689577A0E     (SkyrimSE.exe+1247A0E)         unk_1247960+AE
  [23]  0x7FF689579BF6     (SkyrimSE.exe+1249BF6)         BSScript::SimpleAllocMemoryPagePolicy::Func4_1249B30+C6
  [24]  0x7FF68956710E     (SkyrimSE.exe+123710E)         unk_12370E0+2E
  [25]  0x7FF6895A425B     (SkyrimSE.exe+127425B)         BSScript__Internal__VirtualMachine::unk_12741D0+8B
  [26]  0x7FF689566DDA     (SkyrimSE.exe+1236DDA)         unk_1236D10+CA
  [27]  0x7FF689592A9D     (SkyrimSE.exe+1262A9D)         BSScript__Internal__VirtualMachine::unk_1262760+33D
  [28]  0x7FF689566E9D     (SkyrimSE.exe+1236E9D)         unk_1236E50+4D
  [29]  0x7FF688F33A34     (SkyrimSE.exe+C03A34)          ScrapHeap::Allocate_C034A0+594
  [30]  0x7FF6895749FA     (SkyrimSE.exe+12449FA)         unk_12449D0+2A
  [31]  0x7FF68959A588     (SkyrimSE.exe+126A588)         BSTStaticFreeList_gs_T1_T2_ge_::unk_126A4F0+98
  [32]  0x7FF688F35067     (SkyrimSE.exe+C05067)          unk_C05010+57
  [33]  0x7FF6895A3175     (SkyrimSE.exe+1273175)         BSTCommonLLMessageQueue_gs_T1_ge_::unk_1273120+55
  [34]  0x7FF688F59EB8     (SkyrimSE.exe+C29EB8)          BSFixedString::Release_C29E80+38
  [35]  0x7FF68956B08F     (SkyrimSE.exe+123B08F)         unk_123B040+4F
  [36]  0x7FF6895A415D     (SkyrimSE.exe+127415D)         unk_1274130+2D
  [37]  0x7FF689567F1D     (SkyrimSE.exe+1237F1D)         unk_1237E10+10D
  [38]  0x7FF689566DDA     (SkyrimSE.exe+1236DDA)         unk_1236D10+CA
  [39]  0x7FF688F33E53     (SkyrimSE.exe+C03E53)          ScrapHeap::unk_C03DB0+A3
  [40]  0x7FF688F3CDE3     (SkyrimSE.exe+C0CDE3)          BSSmallBlockAllocator::unk_C0CD60+83
  [41]  0x7FF688F32352     (SkyrimSE.exe+C02352)          Memory::Allocate_C02260+F2
  [42]  0x7FF688F3CDE3     (SkyrimSE.exe+C0CDE3)          BSSmallBlockAllocator::unk_C0CD60+83
  [43]  0x7FF688F32352     (SkyrimSE.exe+C02352)          Memory::Allocate_C02260+F2
  [44]  0x7FF689579D2B     (SkyrimSE.exe+1249D2B)         unk_1249C90+9B
  [45]  0x7FF688462BEA     (SkyrimSE.exe+132BEA)          Mutex::Lock1_132BD0+1A
  [46]  0x7FF6895799E3     (SkyrimSE.exe+12499E3)         BSScript::SimpleAllocMemoryPagePolicy::Func2_12498F0+F3
  [47]  0x7FF689577357     (SkyrimSE.exe+1247357)         unk_1247310+47
  [48]  0x7FF689566E9D     (SkyrimSE.exe+1236E9D)         unk_1236E50+4D
  [49]  0x7FF689572C9F     (SkyrimSE.exe+1242C9F)         unk_1242B00+19F
  [50]  0x7FF689574907     (SkyrimSE.exe+1244907)         unk_12447E0+127
  [51]  0x7FFED183C5BA     (SmartHarvestSE.dll+9C5BA)    
  [52]  0x7FF689592426     (SkyrimSE.exe+1262426)        
  [53]  0x7FF688EEC505     (SkyrimSE.exe+BBC505)          unk_BBC350+1B5
  [54]  0x7FF688EEA417     (SkyrimSE.exe+BBA417)          unk_BB99F0+A27
  [55]  0x7FF688EEB7F3     (SkyrimSE.exe+BBB7F3)          unk_BBB730+C3
  [56]  0x7FF688DBE2EE     (SkyrimSE.exe+A8E2EE)          ahkpWorld::unk_A8E240+AE
  [57]  0x7FF688DCF0C1     (SkyrimSE.exe+A9F0C1)          hkpSimpleConstraintContactMgr::unk_A9EDC0+301
  [58]  0x7FF688EEE86A     (SkyrimSE.exe+BBE86A)          unk_BBE7F0+7A
}
Registers
{
  AX:       0x0                (NULL)
  BX:       0x2241A580730      (BGSKeyword**) -> (Name: VendorItemGem, FormId: 000914ED, File: `Skyrim.esm`)
  CX:       0x223527B2B10      (void*)
  DX:       0x7FFED17DBAAA     (SmartHarvestSE.dll+3BAAA) (void*)
  SI:       0x0                (NULL)
  DI:       0x2241A580740      (void*)
  BP:       0x0                (NULL)
  SP:       0x3EA31F9AF0       (void*)
  IP:       0x7FFED1803640     (SmartHarvestSE.dll+63640) (void*)
  R8:       0x7FFED17A0000     (SmartHarvestSE.dll+0) (char*) "MZ?"
  R9:       0x3EA31F9C10       (void*)
  R10:      0x2227C5085B8      (char*) "Torch"
  R11:      0x3EA31F9AE8       (void*)
  R12:      0x0                (NULL)
  R13:      0x3EA31F9C10       (void*)
  R14:      0x223527B2CB8      (void*)
  R15:      0x224799CE220      (PlayerCharacter**) -> (FormId: 00000014, BaseForm: TESNPC(Name: `Marcus`, FormId: 00000007, File: `NPC Overrides 2020-08-31.esp <- Realm of Lorkhan - Custom Alternate Start - Choose your own adventure.esp <- AI Overhaul.esp <- Skyrim.esm`))
  Flags:    0x10287          
  XMM0:     (double)1.97309442993722E-153 / (float)7.439889E+28
  XMM1:     (double)7.2258796810971E+159 / (float)7.713041E+31
  XMM2:     (double)5.26354424712089E-315 / (float)1
  XMM3:     (double)5.26354424712089E-315 / (float)1
  XMM4:     (double)0 / (float)0
  XMM5:     (double)0 / (float)0
  XMM6:     (double)5.21155998890206E-315 / (float)0.436428
  XMM7:     (double)5.21155998890206E-315 / (float)0.436428
  XMM8:     (double)5.07918525214792E-315 / (float)0.048492
  XMM9:     (double)5.07918525214792E-315 / (float)0.048492
  XMM10:    (double)5.07918525214792E-315 / (float)0.048492
  XMM11:    (double)5.20352094302479E-315 / (float)0.387936
  XMM12:    (double)0 / (float)0
  XMM13:    (double)0 / (float)0
  XMM14:    (double)0 / (float)0
  XMM15:    (double)0 / (float)0
}
Stack
{
  [SP+0]    0x2241A580738      (BGSKeyword**) -> (Name: VendorItemOreIngot, FormId: 000914EC, File: `Skyrim.esm`)
  [SP+8]    0x0                (NULL)
  [SP+10]   0xF                (u8):[15]
  [SP+18]   0x941B8D05592C     (u64):[162846050965804]
  [SP+20]   0x2227D104720      (TESObjectLIGH*) -> (Name: `Torch`, FormId: 00036343, File: `Skyrim.esm`)
  [SP+28]   0x7FFED17F5EF8     (SmartHarvestSE.dll+55EF8) (void*)
  [SP+30]   0x224637E6D60      (void*)
  [SP+38]   0x224799CE220      (PlayerCharacter**) -> (FormId: 00000014, BaseForm: TESNPC(Name: `Marcus`, FormId: 00000007, File: `NPC Overrides 2020-08-31.esp <- Realm of Lorkhan - Custom Alternate Start - Choose your own adventure.esp <- AI Overhaul.esp <- Skyrim.esm`))
  [SP+40]   0x5B0F0A6          (u32):[95482022]
  [SP+48]   0x223527B2B78      (void*)
  [SP+50]   0x2227D104720      (TESObjectLIGH*) -> (Name: `Torch`, FormId: 00036343, File: `Skyrim.esm`)
  [SP+58]   0x224637DB660      (void*)
  [SP+60]   0x2C2              (u16):[706]
  [SP+68]   0x3EA31F9C10       (void*)
  [SP+70]   0x0                (NULL)
  [SP+78]   0x18               (u8):[24]
  [SP+80]   0x7FFED19485E8     (SmartHarvestSE.dll+1A85E8) (void*)
  [SP+88]   0x7FFED17AD02C     (SmartHarvestSE.dll+D02C) (void*)
  [SP+90]   0x224799CE220      (PlayerCharacter**) -> (FormId: 00000014, BaseForm: TESNPC(Name: `Marcus`, FormId: 00000007, File: `NPC Overrides 2020-08-31.esp <- Realm of Lorkhan - Custom Alternate Start - Choose your own adventure.esp <- AI Overhaul.esp <- Skyrim.esm`))
  [SP+98]   0x7FFED18B4F83     (SmartHarvestSE.dll+114F83) (void*)
  [SP+A0]   0x18               (u8):[24]
  [SP+A8]   0x130              (u16):[304]
  [SP+B0]   0x223527B2CB8      (void*)
  [SP+B8]   0x0                (NULL)
  [SP+C0]   0x2228F848C3C      (void*)
  [SP+C8]   0x223527B2CB8      (void*)
  [SP+D0]   0x5B0F0A6          (u32):[95482022]
  [SP+D8]   0x4                (u8):[4]
  [SP+E0]   0x223527B2B10      (void*)
  [SP+E8]   0x224799CE220      (PlayerCharacter**) -> (FormId: 00000014, BaseForm: TESNPC(Name: `Marcus`, FormId: 00000007, File: `NPC Overrides 2020-08-31.esp <- Realm of Lorkhan - Custom Alternate Start - Choose your own adventure.esp <- AI Overhaul.esp <- Skyrim.esm`))
  [SP+F0]   0x0                (NULL)
  [SP+F8]   0x7FFED1818376     (SmartHarvestSE.dll+78376) (void*)
  [SP+100]  0x224799CE220      (PlayerCharacter**) -> (FormId: 00000014, BaseForm: TESNPC(Name: `Marcus`, FormId: 00000007, File: `NPC Overrides 2020-08-31.esp <- Realm of Lorkhan - Custom Alternate Start - Choose your own adventure.esp <- AI Overhaul.esp <- Skyrim.esm`))
  [SP+108]  0x223527B2B78      (void*)
  [SP+110]  0x223527B2B78      (void*)
  [SP+118]  0x221DF9AFFD0      (PlayerCharacter*) -> (FormId: 00000014, BaseForm: TESNPC(Name: `Marcus`, FormId: 00000007, File: `NPC Overrides 2020-08-31.esp <- Realm of Lorkhan - Custom Alternate Start - Choose your own adventure.esp <- AI Overhaul.esp <- Skyrim.esm`))
  [SP+120]  0x7FFED1941370     (SmartHarvestSE.dll+1A1370) (void*)
  [SP+128]  0x223527B2B10      (void*)
  [SP+130]  0xFFFFFFFFFFFFFFFE (i64):[-2]
  [SP+138]  0x0                (NULL)
  [SP+140]  0x2228F848C64      (void*)
  [SP+148]  0x7FF689566E9D     (SkyrimSE.exe+1236E9D) (void*)
  [SP+150]  0xFFFFFFFF00000014 (i64):[-4294967276]
  [SP+158]  0x3EA31F9C10       (void*)
  [SP+160]  0x3EA31FA5C0       (void*)
  [SP+168]  0x0                (NULL)
  [SP+170]  0x22283B7AF30      (void*)
  [SP+178]  0x7FFED1820970     (SmartHarvestSE.dll+80970) (void*)
  [SP+180]  0x14               (u8):[20]
  [SP+188]  0x223527B2B78      (void*)
  [SP+190]  0x224799CE220      (PlayerCharacter**) -> (FormId: 00000014, BaseForm: TESNPC(Name: `Marcus`, FormId: 00000007, File: `NPC Overrides 2020-08-31.esp <- Realm of Lorkhan - Custom Alternate Start - Choose your own adventure.esp <- AI Overhaul.esp <- Skyrim.esm`))
  [SP+198]  0x3EA31F9DA0       (void*)
  [SP+1A0]  0x14               (u8):[20]
  [SP+1A8]  0x7FF6895809E9     (SkyrimSE.exe+12509E9) (void*)
  [SP+1B0]  0x0                (NULL)
  [SP+1B8]  0x14               (u8):[20]
  [SP+1C0]  0x2227C2AC4D0      (BSScript::NativeFunction0<TESObjectREFR, float>*) -> (File: <native>, Type: ObjectReference, Name: GetPositionX)
  [SP+1C8]  0x221DF5F6930      (void*)
  [SP+1D0]  0x3EA31F9CD8       (void*)
  [SP+1D8]  0x2228F848C3C      (void*)
  [SP+1E0]  0x222BE1D3828      (void*)
```

**Comment by SteveTownsend** [comment]
```
  [0]   0x7FFED1803640     (SmartHarvestSE.dll+63640)    
  [1]   0x7FFED17F5EF8     (SmartHarvestSE.dll+55EF8)    
  [2]   0x7FFED17AD02C     (SmartHarvestSE.dll+D02C)     
  [3]   0x7FFED18B4F83     (SmartHarvestSE.dll+114F83)   
  [4]   0x7FFED1818376     (SmartHarvestSE.dll+78376)    
  [5]   0x7FF689566E9D     (SkyrimSE.exe+1236E9D)         unk_1236E50+4D
  [6]   0x7FFED1820970     (SmartHarvestSE.dll+80970)    
```
Mapfile reads this as
```
0001:000623f0       ?_Forced_rehash@?$_Hash@V?$_Uset_traits@U?$pair@II@std@@V?$_Uhash_compare@U?$pair@II@std@@Upair_hash@shse@@U?$equal_to@U?$pair@II@std@@@2@@2@V?$allocator@U?$pair@II@std@@@2@$0A@@std@@@std@@IEAAX_K@Z 00000001800633f0 f i ScanGovernor.obj
 0001:00054e50       ?FilterLootableItems@ContainerLister@shse@@QEAAXV?$function@$$A6A_NPEAVTESBoundObject@RE@@@Z@std@@@Z 0000000180055e50 f   containerLister.obj
0001:0000bfa0       ?_Assign_grow@?$_Hash_vec@V?$allocator@V?$_List_unchecked_const_iterator@V?$_List_val@U?$_List_simple_types@PEAVTESContainer@RE@@@std@@@std@@U_Iterator_base0@2@@std@@@std@@@std@@QEAAX_KV?$_List_unchecked_const_iterator@V?$_List_val@U?$_List_simple_types@PEAVTESContainer@RE@@@std@@@std@@U_Iterator_base0@2@@2@@Z 000000018000cfa0 f i dataCase.obj
 0001:00113f64       ??2@YAPEAX_K@Z             0000000180114f64 f   MSVCRT:new_scalar.obj
0001:000771d0       ?PrepareSPERGMining@papyrus@@YAXPEAUStaticFunctionTag@RE@@@Z 00000001800781d0 f   papyrus.obj
0001:0007f940       ?MarshallAndDispatch@?$NativeFunction@$0A@$$A6AXPEAUStaticFunctionTag@RE@@@ZXPEAU12@$$V@BSScript@RE@@UEBA_NAEAVVariable@23@AEAVVirtualMachine@Internal@23@I0AEBVStackFrame@23@@Z 0000000180080940 f i papyrus.obj
```

**Comment by SteveTownsend** [comment]
This is a _nullptr_ access because the Bound Object under test is a _Torch_. That item has no KYWDs - current SPERG checking assumes they are always there.
```
  [   6]    TESObjectLIGH(Name: `Torch`, FormId: 00036343, File: `Skyrim.esm`)
```
Fix applied for patching into v2.9.6.1
